docs/proposals/20210222-kubelet-authentication.md

```diff
@@ -0,0 +1,331 @@
+
+| OID Suffix | Name                                | Description                                                                                                                                                                                                                            |
+| ---------- | ----------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| 2.1.21     | KubernetesNodeProviderIdentifierOID | An identifier for the machine, should be the same or a derivative of the node provider ID. This is the equivalent of Google’s CloudComputeInstanceIdentifierOID, which we can reuse for a proof of concept (1.3.6.1.4.1.11129.2.1.21). |
+
+#### CSR PEM Blocks
+
+
+
+| Block Name                                 | Description                                          |
+| ------------------------------------------ | ---------------------------------------------------- |
+| KUBELET AUTHENTICATOR ATTESTATION PROVIDER | string describing the attestation provider           |
+| KUBELET AUTHENTICATOR ATTESTATION DATA     | the actual attestation data to perform validation on |
+
+#### attestation data
+
+Attestation data will be appended, with any relevant base64 encoded data wrapped with the following header & footer.
+
+Example CSR:
+```
+-----BEGIN ATTESTATION DATA-----
+S25vd2luZyBtZSBBbGFuIFBhcnRyaWRnZSwga25vd2luZyB5b3UgS3ViZXJuZXRlcyBjbHVzdGVyLCBhaGEh
+-----END ATTESTATION DATA-----
+```
+The format of the attestation block is left to the provider. The Appendix section provides examples.
+
+#### Kubelet serving certificate rotation daemon
+
+The kubelet serving certificate rotation daemon runs as a systemd or windows service manager service on the node. It checks for the existence of the serving certificate files, their validity (e.g. not self signed), or request a new serving certificate. Once the new serving certificate is written, kubelet can be restarted.
+
+#### Core Specification
+- Core Cluster API MUST provide the following implementations of CSRs and signers:
+    - `cluster.x-k8s.io/kube-apiserver-client-kubelet-insecure` which implement an “Always Allow” type signer that provides equivalent security to Cluster API v1alpha3. This is only to be used for providers where no secure mechanism exists.
+    - `cluster.x-k8s.io/kube-apiserver-client-kubelet-tpm` and `cluster-x-k8s-io/kubelet-serving-tpm`
+        - Will implement TPM-based certificate signers and requesters based on the cloud-provider-gcp implementation.
+        - However, since the mechanism for retrieving endorsement keys varies across platforms, the TPM signer will additionally require a provider specific mechanism to provide the TPM Endorsement Key.
+
+#### Provider Specification
+
+#### All providers
```

Comment:
i think the providers here should be on the `#####` level

Reply:
Fixed. All of these should have been under implementation details.